### ISSUE TITLE
[ArangoDB Server 3.1.6] Unsupported content type: client.DatabaseExists when Database already exists

### ISSUE THREAD

**Issue by SoufienMIMS** [arangodb/go-driver#36]
Hi,

I began to use your GoLang Driver for ArangoDB and I realized that I got an awkward error when using the call DatabaseExists.

When a database does not exist and we ask for its existence, the behavior is totally normal and return _false, nil_.

When a database does exist and we ask for its existence, we have this error :  _Unsupported content type: text/plain; charset=utf-8_.

I'm using actually this configuration

> configHTTP := http.ConnectionConfig{
> 		Endpoints:   []string{"127.0.0.1:4242"},
> 		TLSConfig:   &tls.Config{InsecureSkipVerify: true},
> 		ContentType: driver.ContentTypeJSON}

with a BasicAuthentication

Are you familiar with this issue?

Soufien

EDIT:

I am working with an ArangoDB server in a Docker container deployed in a Microsoft Azure VM.

I realized with my local ArangoDB server (3.1.21), it works pretty fine but on the Docker Image server (3.1.6) this issue occurs. Does it make sense to you ?

**Comment by ewoutp** [comment]
Thanks for reporting this issue. Similar error has been seen before (#35) but we cannot reproduce it. We'll try again with the 3.1.6 version and get back to you.

**Comment by SoufienMIMS** [comment]
Good, thanks for your reactivity. I'll be waiting for your feedback.

Soufien

**Comment by ewoutp** [comment]
I've tried the 3.1.6 docker image and that works fine.
Please try the following:

- Run arangodb 3.1.6 docker image locally (see if it works locally)
- If it does work locally, please describe what there is between the Arango database on Azure and your client. If this a direct connection? Any load-balancers/proxies in between?

**Comment by SoufienMIMS** [comment]
Hi Ewout,

I've tried with a local server, it works fine on my MacOSX (16.5.0 Darwin Kernel Version 16.5.0).

My Virtual machine on Azure is on a Linux System (Ubuntu 16.04.1 LTS Kerne 4.4.0-78). We have a direct connection to the ArangoDB server without Load Balancing or Proxies. My client is on a MaxOSX (16.5.0 Darwin Kernel Version 16.5.0).

Do you need any other informations that could help it ?

**Comment by ewoutp** [comment]
Thx. I'm forwarding the problem internally. I'll get back as soon as I know more.

**Comment by ewoutp** [comment]
It seems that there are some case where this content type is returned. Don't understand why yet.

Can you try the driver in this branch and report the error that you get.
https://github.com/arangodb/go-driver/tree/unsupported-content-type

It should provide more context info.

**Comment by SoufienMIMS** [comment]
I got this error : 

> [ERROR] Unsupported HTTP response content type: text/plain; charset=utf-8, status: 401, body:

That sounds like I'm not having the right authorization. Seems awkward because I'm able to create a Database but not to ask if it exists. Does this clarify the situation for you ?

**Comment by ewoutp** [comment]
What are the access settings on that database?

**Comment by SoufienMIMS** [comment]
I've just seen on the interface that I didn't had the access granted for writing / read for the collection that I generated thanks to the driver.

I forgot to add Granted access to my user thanks to this line : 

> user.GrantReadWriteAccess(nil, database)

I though the access was granted implicitly when a user create a DB with its credential. I added this command above to "force" the access to the database.

Will we have a more precise feedback in the future versions ?

Many thanks for you help and your time,
